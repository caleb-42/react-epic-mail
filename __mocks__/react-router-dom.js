/* eslint-disable react/display-name */
/* eslint-disable react/prop-types */
import React from 'react';

const rrd = require('react-router-dom');

rrd.HashRouter = ({ children }) => <div>{children}</div>;


module.exports = rrd;