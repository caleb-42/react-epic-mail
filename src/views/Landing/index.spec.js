import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from 'react-router-dom';
import App from "@routes/app";

it("Landing page", () => {
  const tree = renderer.create(
    <MemoryRouter initialEntries={['/']}>
      <App />
    </MemoryRouter>
  );
  expect(tree).toMatchSnapshot();
});
