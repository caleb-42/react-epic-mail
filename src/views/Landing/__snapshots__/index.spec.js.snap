// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing page 1`] = `
<div>
  <div
    className="landingPage h-full overflow-auto"
  >
    <nav
      className="appNavig  anim"
    >
      <div
        className="flex md:flex-1 w-full"
      >
        <div
          className="flex-1 flex justify-between items-center"
        >
          <a
            className="logo"
            href="#"
          />
        </div>
        <button
          className="hamburger hamburger--collapse   md:hidden"
          onClick={[Function]}
          type="button"
        >
          <span
            className="hamburger-box"
          >
            <span
              className="hamburger-inner"
            />
          </span>
        </button>
      </div>
      <div
        className="md:flex md:items-center md:w-auto w-full"
        id="menu"
      >
        <div>
          <ul
            className="md:flex items-center justify-between text-base text-white font-montserrat font-medium pt-4 md:pt-0"
          >
            <li>
              <a
                className="md:p-4 py-3 px-0 block border-b-2 border-transparent hover:border-yellow-400"
                href="/"
                onClick={[Function]}
              >
                Features
              </a>
            </li>
            <li>
              <a
                className="md:p-4 py-3 px-0 block border-b-2 border-transparent hover:border-yellow-400 md:mb-0 mb-2"
                href="/"
                onClick={[Function]}
              >
                Support
              </a>
            </li>
            <li>
              <a
                className="md:p-4 py-3 px-0 block border-b-2 border-transparent hover:border-yellow-400"
                href="/signup"
                onClick={[Function]}
              >
                Sign Up
              </a>
            </li>
            <li>
              <a
                className="md:p-4 py-3 px-0 block border-b-2 border-transparent hover:border-yellow-400"
                href="/login"
                onClick={[Function]}
              >
                Log In
              </a>
            </li>
          </ul>
        </div>
        <a
          className="hidden md:flex md:ml-4 items-center justify-center md:mb-0 mb-4 pointer-cursor"
          href="#"
        >
          <img
            alt="Andy Leverenz"
            className="rounded-full w-10 h-10 border-2 border-transparent hover:border-indigo-400"
            src="/dp.png"
          />
        </a>
      </div>
    </nav>
    <div
      className="z-20 relative"
    >
      <div>
        <div
          data-src="emailing.png"
        >
          <div
            className="absolute md:ml-32 sm:ml-18 ml-8 mt-48 w-64 top-0 left-0"
          >
            <p
              className="md:text-3xl text-xl sm:text-2xl inline-block pr-4 border-green-400 border-b-4 text-white"
            >
              Send SMS
            </p>
            <p
              className="md:text-6xl text-4xl sm:text-5xl text-white"
            >
              Messages
            </p>
          </div>
        </div>
        <div
          data-src="landing-sm.png"
        >
          <div
            className="absolute md:ml-32 sm:ml-18 ml-8 mt-48 w-64 top-0 left-0"
          >
            <p
              className="md:text-3xl text-xl sm:text-2xl inline-block pr-4 border-green-400 border-b-4 text-white"
            >
              Snap & Upload
            </p>
            <p
              className="md:text-6xl text-4xl sm:text-5xl text-white"
            >
              Pictures
            </p>
          </div>
        </div>
        <div
          data-src="landing-desktop.png"
        >
          <div
            className="absolute md:ml-32 sm:ml-18 ml-8 mt-48 w-64 top-0 left-0"
          >
            <p
              className="md:text-3xl text-xl sm:text-2xl inline-block pr-4 border-green-400 border-b-4 text-white"
            >
              Stay
            </p>
            <p
              className="md:text-6xl text-4xl sm:text-5xl text-white"
            >
              Connected
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="w-full pt-12 pb-20 bg-white offers relative z-10"
    >
      <div
        className="lg:w-2/5 md:3/5 w-4/5 mx-auto"
      >
        <h1
          className="text-center text-gray-800 w-100 text-2xl mb-4 font-medium"
        >
          What we offer
        </h1>
        <div
          className="w-full mx-auto mb-8 flex flex-col sm:flex-row"
        >
          <div
            className="flex-1 inline-block"
          >
            <div
              className="pointer anim w-full"
            >
              <div
                className="Innovation"
              />
            </div>
            <h3
              className="text-center w-full"
            >
              Innovation
            </h3>
          </div>
          <div
            className="flex-1 inline-block"
          >
            <div
              className="pointer anim w-full"
            >
              <div
                className="Speed"
              />
            </div>
            <h3
              className="text-center w-full"
            >
              Speed
            </h3>
          </div>
          <div
            className="flex-1 inline-block"
          >
            <div
              className="pointer anim w-full"
            >
              <div
                className="Support"
              />
            </div>
            <h3
              className="text-center w-full"
            >
              Support
            </h3>
          </div>
        </div>
        <p
          className="w-100 text-center text-gray-600 text-base"
        >
          Epic mail is a sleek, efficient mailing application capable of having all your messaging needs met.
        </p>
      </div>
    </div>
    <div
      className="features bg-white relative"
    >
      <div
        className="title mx-auto bg-gray-600 md:w-1/5 w-3/5 py-2 rounded-full shadow-lg -mt-8 relative z-20"
      >
        <h3
          className="text-center text-white text-2xl font-medium"
        >
          Features
        </h3>
      </div>
      <div
        className="mx-auto lg:w-3/4 md:w-4/5 w-full py-12"
      >
        <div
          className="pointer sm:w-2/4 w-4/5 lg:px-16 md:px-8 py-8 align-middle md:inline-block block mx-auto text-center anim"
        >
          <div
            className="w-4/5 mx-auto"
          >
            <div
              className="devices"
            />
            <div
              className="text-center"
            >
              <h2
                className="opac-70 text-lg font-medium my-2"
              >
                Control from any device
              </h2>
              <p
                className="opac-50 text-sm text-gray-700"
              >
                The app is allows you view and send mails from your phones, tablets and desktop devices
              </p>
            </div>
          </div>
        </div>
        <div
          className="pointer sm:w-2/4 w-4/5 lg:px-16 md:px-8 py-8 align-middle md:inline-block block mx-auto text-center anim"
        >
          <div
            className="w-4/5 mx-auto"
          >
            <div
              className="upload"
            />
            <div
              className="text-center"
            >
              <h2
                className="opac-70 text-lg font-medium my-2"
              >
                Picture upload
              </h2>
              <p
                className="opac-50 text-sm text-gray-700"
              >
                Place an image to that wonderful reputation of yours by Uploading a display picture
              </p>
            </div>
          </div>
        </div>
        <div
          className="pointer sm:w-2/4 w-4/5 lg:px-16 md:px-8 py-8 align-middle md:inline-block block mx-auto text-center anim"
        >
          <div
            className="w-4/5 mx-auto"
          >
            <div
              className="reset"
            />
            <div
              className="text-center"
            >
              <h2
                className="opac-70 text-lg font-medium my-2"
              >
                Password reset
              </h2>
              <p
                className="opac-50 text-sm text-gray-700"
              >
                Ever forgot your password and wondered how you could reset it... Epic mail to the rescue.
              </p>
            </div>
          </div>
        </div>
        <div
          className="pointer sm:w-2/4 w-4/5 lg:px-16 md:px-8 py-8 align-middle md:inline-block block mx-auto text-center anim"
        >
          <div
            className="w-4/5 mx-auto"
          >
            <div
              className="sms"
            />
            <div
              className="text-center"
            >
              <h2
                className="opac-70 text-lg font-medium my-2"
              >
                SMS Messaging
              </h2>
              <p
                className="opac-50 text-sm text-gray-700"
              >
                Get sms messages as an add on to your mail sending services
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer
      className=" text-center text-sm py-3 w-full"
    >
      <h3
        className="text-white "
      >
        EPICMAIL Copyright 
        2019
      </h3>
    </footer>
  </div>
  <div
    className="Toastify"
  />
</div>
`;
